### DISCUSSION TITLE
MacOS 12 (beta) + Dark Theme + v204

### DISCUSSION THREAD

**Discussion by omarsultan** [collin80/SavvyCAN#406] [Q&A]
With the above combo, when I turn on "interpret frames", the text is black and illegible on the black background:

<img width="1656" alt="image" src="https://user-images.githubusercontent.com/6629296/127561395-f8610880-5bd8-4fa4-b337-2999999eec36.png">

Assume this is not the way it is supposed to be.  I am new to SavvyCAN, so is this a SavvyCAN issue or something with my set-up?

**Comment by collin80** [comment]
It's a SavvyCAN issue. A lot of signals get defaulted to black text. It's supposed to make the defaults whatever the system sets. In that case it should have been black background and white text. But, sometimes DBC files get saved with black text and it picks that up and you end up with black on black which isn't all that legible to say the least. I need to better detect stupidity like that and force some common sense when things like this happen. Really, it should never allow black on black or white on white (or really any color combo where the text and background are very close to the same color).

That all might get a bit complicated but I could probably more quickly add a configuration option that forces system colors onto signals so that it directly overrides everything and "does the right thing" then circle back and do it all properly later on.

**Comment by zbynekwinkler** [comment]
I have created a new dbc using SavvyCAN and not knowing otherwise, I have not changed any color of anything which resulted in this all black screen which is how I got here. All I needed to do was to change the background color of my messages to white. So if the default color could be changed from black on black to black on white, I would have never noticed a problem.

**Comment by brainstorm** [comment]
There are other dark theme bugs in other windows, see:

<img width="633" alt="Screenshot 2023-05-20 at 3 59 42 pm" src="https://github.com/collin80/SavvyCAN/assets/175587/d9740196-607b-4e7e-94c4-12579525a799">
<img width="626" alt="Screenshot 2023-05-20 at 4 00 08 pm" src="https://github.com/collin80/SavvyCAN/assets/175587/7aa2704b-e264-4197-9895-208f6f1a1f41">
